src/internet/radiobrowser/radiobrowserservice.cpp

```diff
@@ -419,9 +420,28 @@ void RadioBrowserService::ResolveStationUrl(const QUrl& original_url) {
     reply->deleteLater();
     QJsonDocument document = ParseJsonReply(reply);
 
-    QJsonObject item = document.object();
+    StreamList list;
+    QJsonArray contents = document.array();
+    QJsonValue item = contents.first();
+    if (item.isUndefined()) return;
+
+    Song ret;
+    ret.set_valid(true);
+    ret.set_title(item["name"].toString());
     QUrl url(item["url"].toString());
-    emit StationUrlResolved(original_url, url);
+    ret.set_url(url);
+    ret.set_art_automatic(item["favicon"].toString());
+
+    emit StreamMetadataFound(original_url, ret);
+    emit StationUrlResolved(original_url, ret.url());
```

Comment:
Would it make sense to emit one signal with the metadata and let the URL handler extract the URL?

Reply:
Yes, the double emit looks fishy to me, too, but  as far as i understand `StreamMetadataFound` can only be emitted by subclasses of `InternetModel` and `InternetService`.
Maybe metadata could be injected inside the `LoadResult` object by the url handler?

Reply:
I was thinking that this would just emit StreamMetadataFound and RadioBrowserUrlHandler connect to that signal. But, again, I think the fact that you need a signal in the error case above makes this a less reasonable solution.

Reply:
Reworked using StreamMetadataFound for success and another signal only for failures.